conseil-common/src/main/scala/tech/cryptonomic/conseil/common/cache/MetadataCaching.scala

```diff
@@ -68,32 +76,36 @@ class MetadataCaching[F[_]](
     cachingStatus.update(_ => status)
 
   /** Reads entities from cache */
-  def getEntities: String => F[Option[CacheEntry[List[Entity]]]] =
+  def getEntities: EntityCacheKey => F[Option[CacheEntry[List[Entity]]]] =
     getFromCache(entitiesCache)
 
   /** Reads attributes from cache for given entity */
-  def getAttributes: String => F[Option[CacheEntry[List[Attribute]]]] =
+  def getAttributes: AttributeCacheKey => F[Option[CacheEntry[List[Attribute]]]] =
     getFromCache(attributesCache)
 
   /** Generic method for getting value from cache */
-  private def getFromCache[A](cache: MVar[F, Cache[A]])(key: String): F[Option[CacheEntry[A]]] =
-    cache.read.map(_.get(CacheKey(key)))
+  private def getFromCache[K <: CacheKey, A](cache: MVar[F, Cache[K, A]])(key: K): F[Option[CacheEntry[A]]] =
+    cache.read.map(_.get(key))
 
   /** Reads all attributes from cache */
   def getAllAttributes: F[AttributesCache] =
     attributesCache.read
 
+  /** Reads all attributes from cache for given seq of keys */
+  def getAllAttributesByKeys: Seq[AttributeCacheKey] => F[AttributesCache] =
```

Comment:
scala `Set` can be used "intentionally" as a function that verifies if an element belongs.

This means that we could use it as input `Set[AttributeCacheKey] => F[AttfributesCache]`
to use it like `attributesCache.read.map(_.filterKeys(keys))`

as `keys <: AttributeCacheKey => Boolean`

Would you think it's too obscure a use?

Reply:
I don't think so. I believe we can use that actually and that should not be a big deal. Remember that we need to transform list to set before passing it here, since originally, we are dealing with list. That is the reason of the `keys.contains` function here :)